### PR TITLE
Feature one to Develop

### PR DESCRIPTION
**Pull request by DickensJuma**
### Authentication and Posts
- Use Device gem for user authentication
- Create a Post model and a Posts controller and a corresponding resource 
- Atop  Posts Controller, create the necessary helper methods in the ApplicationController.
- Write a very simple form that creates a new Post.
- Made your corresponding `#create` action build a post where the foreign key for the author is automatically populated based on whichever user is signed in. 
- Redirect to the Index view if successful.
- Fill out the`#index` action of the PostsController and create the corresponding view, showing a list of every post.
- Display the author’s name, but only if a user is signed in.

### CODE REVIEW

**Review by davidmrtz-dev — CHANGES_REQUESTED** [review]
## Hi Team  👨‍🚀 
### Very good job so far.

However, there are some issues that need to be addressed before moving on.  👨‍💻  ⬇️ 
Please check out below.

****
- [x] Readme.
Actually your readme file is pretty good, but there's a little detail on the "how to use" section, specifically in the last command, the user should want to get the server running instead of the console, but you can specify each one and let the user choose.
****
- [x] As a signed-in user you can see post's author ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/signed_in_can_see_author_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))
As shown below as a signed-in user I can't see the authors of the posts.

Screen Shot 2020-08-13 at 11 11 15 PM

Please fix this adding the required label on the index posts views.
Note = Remember all the logic should be into the controller not in the view, so you have to determine if the user is signed in the controller and according to that procced to show or hide the authors.
Consider the following example:

```ruby
@posts = Post.all
    unless user_signed_in?
      @posts.each do |post|
        post.user.username = 'Anonymous'
      end
    end
@posts
```

****
- [x] It creates the application routes using resources :posts, only: [:new, :create, :index] (screenshot from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))



You have this on your routes:

Screen Shot 2020-08-13 at 11 20 06 PM

Kindly update your posts resources according to the given requirements.

Consider the following example

```ruby
resources :posts, only: %i[create new index]
```

****

### **Please make the changes required and submit for re-review once you are ready!** 

****

### Status: Changes required. ♻️  
****

Happy coding!  💻  
Have a great day 🙂

If you need more clarifications kindly reach me on slack.
**[TSE:  David Elí]** 
Slack: @ David Eli Martinez Garcia

**Comment by Anna-Myzukina** [comment]
## Hi team :clap: 

### You did a great job so far! :+1:  , just a few changes before it can be approved:

##### You should implement Posts Controller


- [ ] it uses before_action to restrict access to the `:new` and `:create` methods (uses ApplicationController helper method) 
screenshot 


### [OPTIONAL]
- ["How to Write Good Commit Messages: A Practical Git Guide"](https://www.freecodecamp.org/news/writing-good-commit-messages-a-practical-guide/)
```
Write your commit message in the imperative and capitalize: "Fix linter" and not "fixed bug"
or "Fixes bug."  This convention matches up with commit messages generated
by commands like git merge and git revert.
```

### Kindly make the changes and submit a new code review request.
Thanks!

*If you have any questions, let me know on slack (@Anna-Muzykina) and I will be glad to clarify with you.*
*Happy coding * <g-emoji class="g-emoji" alias="computer" fallback-src="https://github.githubassets.com/images/icons/emoji/unicode/1f4bb.png">computer</g-emoji>

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -0,0 +1,81 @@
+class PostsController < ApplicationController
+  before_action :set_post, only: %i[show edit update destroy]
+  before_action :authenticate_user!, except: %i[index show]
```
Comment (Anna-Myzukina) [inline app/controllers/posts_controller.rb:3]: `before_action :signed_in_only!, only: [:new, :create]`

**Comment by Anna-Myzukina** [comment]
## [APPROVED]  :heavy_check_mark: 
### You did a great job so far :+1: 

### Go and merge.

# :raising_hand:  See you at the next projects.

Happy coding  <g-emoji class="g-emoji" alias="computer" fallback-src="https://github.githubassets.com/images/icons/emoji/unicode/1f4bb.png">computer</g-emoji>

**Pull request merged**